pkg/manager/manager.go

```diff
@@ -1890,11 +1890,69 @@ func (m *jobManager) CheckValidJobConfiguration(req *JobRequest) error {
 	return nil
 }
 
+// validVersionRegexes represents all the valid version formats we recognize for OpenShift.
+// We allow an optional comma in the front so we can account for both 4.19,xxxx and xxxx,4.19
+var validVersionRegexes = []*regexp.Regexp{
+	regexp.MustCompile(`^\d+\.\d+`),               // 4.19
+	regexp.MustCompile(`^nightly`),                // nightly
+	regexp.MustCompile(`^ci`),                     // ci
+	regexp.MustCompile(`^\d+\.\d+\.nightly`),      // 4.19.nightly
+	regexp.MustCompile(`^\d+\.\d+\.ci`),           // 4.19.ci
+	regexp.MustCompile(`^\d+\.\d+\.0-0\.nightly`), // 4.19.0-0.nightly
+	regexp.MustCompile(`^\d+\.\d+\.0-0\.ci`),      // 4.19.0-0.ci
+
+	// Release specs for nightly and ci
+	regexp.MustCompile(`^.*release:\d+\.\d+\.\d+-0\.nightly-\d{4}-\d{2}-\d{2}-\d{6}$`),
+	regexp.MustCompile(`^.*release:\d+\.\d+\.\d+-0\.ci-\d{4}-\d{2}-\d{2}-\d{6}$`),
```

Comment:
I'm concerned that this is a little loose...
What's to stop someone for asking for: `docker.io/ocp/release:4.20.0-0.ci-2025-04-25-034753`?

Reply:
something like `docker.io/ocp/release:4.20.0-0.ci-2025-04-25-034753` will result in an error as in this failed [job](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/release-openshift-origin-installer-launch-aws-modern/1916890116282912768).

If the container registry will only be `quay.io` and `registry.ci.openshift.org` then we can restrict it to just those so we don't even get to the part where the prow job attempts to find the image.

Reply:
Yes, that would be ideal!  We don't want to waste resources on anything but what we provide.